src/search-js/decorator.ts

```diff
@@ -0,0 +1,49 @@
+import { SearchProduct, SearchProductSku } from "@nosto/nosto-js/client"
+import { CurrencyConfig, getCurrencyFormatting } from "./currencies"
+
+type Prices = Pick<SearchProduct, "price" | "listPrice">
+
+type FormattedPrices = {
+  priceText?: string
+  listPriceText?: string
+}
+
+/**
+ * Exposes currency formatting logic as a SearchProduct decorator
+ */
+export function priceDecorator(config?: Partial<CurrencyConfig>) {
+  const { formatCurrency } = getCurrencyFormatting(config)
+
+  function formatPrices<T extends Prices>(obj: T, currency?: string) {
+    const formatted: FormattedPrices = {}
+    if (obj.price !== undefined) {
+      formatted.priceText = formatCurrency(obj.price, currency)
+    }
+    if (obj.listPrice !== undefined) {
```

Comment:
Also, can this be null?

Reply:
yes, if the listPrice is not mapped in the request
all SerchProduct fields are optional

Reply:
Should we also make sure this value is not null then?

Reply:
sorry, I misunderstood. Actual null would be broken data. I think handling undefined should be sufficient here.
Null would mean that it has been set to null

Reply:
Based on the type definition price & listPrice are optional number typed fields
null would be illegal

Reply:
Fair enough :)